prometheus_test.go

```diff
@@ -448,3 +449,88 @@ tests_foo{key_1="issue659",key_2="",key_3="issue659",key_4="",key_5="issue659"}
 		t.Fatalf("output differed from expected output: %s want: %s", output, want)
 	}
 }
+
+func TestShareDefaultRegistry(t *testing.T) {
+	_, err := NewExporter(Options{
+		Registerer: prometheus.DefaultRegisterer,
+		Gatherer:   prometheus.DefaultGatherer,
+	})
+	if err != nil {
+		t.Fatalf("failed to create prometheus exporter: %v", err)
+	}
+	m := stats.Int64("tests/foo", "foo", stats.UnitDimensionless)
+	v := &view.View{
+		Name:        m.Name(),
+		Description: m.Description(),
+		Measure:     m,
+		Aggregation: view.Count(),
+	}
+	if err := view.Register(v); err != nil {
+		t.Fatalf("failed to create views: %v", err)
+	}
+	defer view.Unregister(v)
+	view.SetReportingPeriod(time.Millisecond)
```

Comment:
since prometheus exporter is using new metricexport.Reader interface it simply reads when request is made to /metrics endpoint. So setting reporting period is not necessary. This was required before when Oc library used to push metrics to prometheus exporter.